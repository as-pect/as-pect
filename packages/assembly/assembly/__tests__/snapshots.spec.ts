class Vec3 {
  public constructor(
    public x: f64 = 0,
    public y: f64 = 0,
    public z: f64 = 0,
  ) {}
}

describe("a set of snapshots", () => {
  test("some snapshots", () => {
    expect(1).toMatchSnapshot("an integer");
    expect(3.14).toMatchSnapshot("a float");

    expect([1, 2, 3, 4, 5]).toMatchSnapshot("an integer array");

    let a = new Float64Array(4);
    a[0] = 1;
    a[1] = 2;
    a[2] = 3;
    a[3] = 4;
    expect(a).toMatchSnapshot("a typed array");
    expect("a string").toMatchSnapshot("a string");
    expect(new Vec3(1, 2, 3)).toMatchSnapshot("a reference");
  });
});
