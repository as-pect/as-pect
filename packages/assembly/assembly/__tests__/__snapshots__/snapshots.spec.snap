exports[`some snapshots!~an integer[0]`] = `1`;

exports[`some snapshots!~a float[0]`] = `3.14`;

exports[`some snapshots!~an integer array[0]`] = `Array<i32> [
  1,
  2,
  3,
  4,
  5
]`;

exports[`some snapshots!~a typed array[0]`] = `Float64Array [ 1.0, 2.0, 3.0, 4.0 ]`;

exports[`some snapshots!~a string[0]`] = `"a string"`;

exports[`some snapshots!~a reference[0]`] = `Vec3 {
  x: 1.0,
  y: 2.0,
  z: 3.0
}`;
